rasterio/_io.pyx

```diff
@@ -1063,6 +1077,54 @@ cdef class DatasetReaderBase(DatasetBase):
         # generator implemented in sample.py.
         return sample_gen(self, xy, indexes=indexes, masked=masked)
 
+    def statistics(self, bidx, approx=False, clear_cache=False):
+        """Get min, max, mean, and standard deviation of a raster band.
+
+        Parameters
+        ----------
+        bidx : int
+            The band's index (1-indexed).
+        approx : bool, optional
+            If True, statistics will be calculated from reduced
+            resolution data.
+        clear_cache : bool, optional
+            If True, saved stats will be deleted and statistics will be
+            recomputed. Requires GDAL version >= 3.2.
+
+        Returns
+        -------
+        Statistics
+
+        Notes
+        -----
+        GDAL will preferentially use statistics kept in raster metadata
+        like images tags or an XML sidecar. If that metadata is out of
+        date, the statistics may not correspond to the actual data.
+
+        Additionally, GDAL will save statistics to file metadata as a
+        side effect if that metadata does not already exist.
+
+        """
+        cdef double min, max, mean, std
+        cdef GDALRasterBandH band = NULL
+
+        band = self.band(bidx)
+
+        if clear_cache:
+            IF (CTE_GDAL_MAJOR_VERSION, CTE_GDAL_MINOR_VERSION) >= (3, 2):
+                GDALDatasetClearStatistics(self._hds)
+            ELSE:
```

Comment:
I think that there isn't a need for `ELSE` here.

Reply:
One idea I had was to warn that the option was unused in the `ELSE` block instead of `pass`.